flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -309,6 +313,19 @@ public static ServerSocket createSocketFromPorts(Iterator<Integer> portsIterator
 		return null;
 	}
 
+	/**
+	 * Returns the wildcard address to listen on all interfaces.
+	 * @return Either 0.0.0.0 or :: depending on the IP setup.
+	 */
+	public static String getWildcardIPAddress() {
```

Comment:
You could move that code into a static initializer. That way you don't need to lock and check if it was set before.